testing/sdk_install.py

```diff
@@ -164,77 +186,44 @@ def _uninstall(
-
-        log.info(
-            'Uninstall done after pkg({}) + janitor({}) = total({})'.format(
-                shakedown.pretty_duration(janitor_start - start),
-                shakedown.pretty_duration(finish - janitor_start),
-                shakedown.pretty_duration(finish - start)))
+            run_janitor(service_name, role, service_account, zk)
+        except Exception as e:
+            log.info('Got exception when trying to janitor {}'.format(service_name))
+            log.info(traceback.format_exc())
+            raise
     else:
-        log.info('Uninstalling {}'.format(service_name))
+        log.info('Cleaning up {}'.format(service_name))
         try:
-            shakedown.uninstall_package_and_wait(
-                package_name, service_name=service_name)
-            # service_name may already contain a leading slash:
-            marathon_app_id = '/' + service_name.lstrip('/')
-            log.info('Waiting for no deployments for {}'.format(marathon_app_id))
-            shakedown.deployment_wait(TIMEOUT_SECONDS, marathon_app_id)
-
-            # wait for service to be gone according to marathon
-            client = shakedown.marathon.create_client()
-            def marathon_dropped_service():
-                app_ids = [app['id'] for app in client.get_apps()]
-                log.info('Marathon apps: {}'.format(app_ids))
-                matching_app_ids = [
-                    app_id for app_id in app_ids if app_id == marathon_app_id
-                ]
-                if len(matching_app_ids) > 1:
-                    log.warning('Found multiple apps with id {}'.format(
-                        marathon_app_id))
-                return len(matching_app_ids) == 0
-            log.info('Waiting for no {} Marathon app'.format(marathon_app_id))
-            shakedown.time_wait(marathon_dropped_service, timeout_seconds=TIMEOUT_SECONDS)
-
-        except (dcos.errors.DCOSException, ValueError) as e:
-            log.info(
-                'Got exception when uninstalling package: {}'.format(e))
-            if 'marathon' in str(e):
-                log.info('Detected a probable marathon flake. Raising so retry will trigger.')
-                raise
-        finally:
-            sdk_utils.list_reserved_resources()
+          sdk_marathon.wait_for_deployment_and_app_removal(
+              sdk_marathon.get_app_id(service_name), timeout=TIMEOUT_SECONDS)
+        except Exception as e:
+            log.info('Got exception when waiting for service app to be removed {}'.format(service_name))
+            log.info(traceback.format_exc())
+            raise
+
+    finish = time.time()
+
+    log.info(
+        'Uninstalled {} after pkg({}) + cleanup({}) = total({})'.format(
+            service_name,
+            shakedown.pretty_duration(cleanup_start - start),
+            shakedown.pretty_duration(finish - cleanup_start),
+            shakedown.pretty_duration(finish - start)))
+
+    sdk_utils.list_reserved_resources()
```

Comment:
The fact that this is now not in a `finally` block subtly changes how this will behave in the presence of an exception.